chunk_downloader.go

```diff
@@ -64,6 +65,11 @@ type snowflakeChunkDownloader struct {
 	FuncGet            func(context.Context, *snowflakeChunkDownloader, string, map[string]string, time.Duration) (*http.Response, error)
 }
 
+type arrowChans struct {
+	mu sync.Mutex
```

Comment:
why do we have a mutex?

Reply:
as briefly chatted: I had pretty frequent "[concurrent map](https://stackoverflow.com/questions/45585589/golang-fatal-error-concurrent-map-read-and-map-write/45585833)" action errors - Mutex solves this (briefly lock and unlock the object)

Reply:
changing this to sync.Map!